pkg/controller/apimanager/apimanager_controller.go

```diff
@@ -100,11 +104,50 @@ type ReconcileAPIManager struct {
 	operator.BaseControllerReconciler
 }
 
+const (
+	ThreescaleVersionAnnotation = "apps.3scale.net/apimanager-threescale-version"
+	OperatorVersionAnnotation   = "apps.3scale.net/threescale-operator-version"
+)
+
+func (r *ReconcileAPIManager) updateVersionAnnotations(cr *appsv1alpha1.APIManager) error {
+	if cr.Annotations == nil {
+		cr.Annotations = map[string]string{}
+	}
+	cr.Annotations[ThreescaleVersionAnnotation] = product.ThreescaleRelease
+	cr.Annotations[OperatorVersionAnnotation] = version.Version
+	err := r.Client().Update(context.TODO(), cr)
+	return err
+}
+
+func (r *ReconcileAPIManager) upgradeAPIManager(cr *appsv1alpha1.APIManager) (reconcile.Result, error) {
+	// The object to instantiate would change in every release of the operator
+	// that upgrades the threescale version
+	upgrade := Upgrade25_to_26{
```

Comment:
upgrade25_to_26 does not make sense

Reply:
:+1: yes at the end the struct name will be another one for sure.
This was just only an example to test the mechanism when 2.6 was not released yet.